ext/RastersProjExt/reproject.jl

```diff
@@ -0,0 +1,53 @@
+
+Rasters._reproject(source::GeoFormat, target::GeoFormat, dim::XDim, vals::AbstractVector) = _reproject!(source, target, dim, Float64.(vals))
+Rasters._reproject(source::GeoFormat, target::GeoFormat, dim::YDim, vals::AbstractVector) = _reproject!(source, target, dim, Float64.(vals))
+
+function _reproject!(source::GeoFormat, target::GeoFormat, ::XDim, vals::Vector{Float64}) 
+    # First, construct a transformation from `source` to `target`.
+    # TODO: add area of use, so that the transformation is more accurate.
+    trans = Proj.Transformation(source, target; always_xy = true, direction = Proj.PJ_FWD)
```

Comment:
We can't just dispatch on GeoFormat because Proj.Transformation dispatches on concrete types and source and target need to be the same type.

Reply:
Proj doesn't require that, both CRS are wrapped in Proj.CRS and then it goes to the default constructor. I just fixed the dispatch there to allow WellKnownText too.

Reply:
Ah I just misunderstood the test fails then. So we just need a patch bump for Proj and it will work?

Reply:
I think so - we should also apply that change here.

Reply:
Reproject tests pass now, looks like that was all we needed at this point.